src/renderer/components/+catalog/catalog-entity-detail-items/extension-catalog-entity-detail-items-registrator.injectable.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { pipeline } from "@ogre-tools/fp";
+import { getInjectable } from "@ogre-tools/injectable";
+import { forEach } from "lodash/fp";
+import { extensionRegistratorInjectionToken } from "../../../../extensions/extension-loader/extension-registrator-injection-token";
+import type { LensRendererExtension } from "../../../../extensions/lens-renderer-extension";
+import { catalogEntityDetailItemInjectionToken } from "./catalog-entity-detail-item-injection-token";
+
+const extensionCatalogEntityDetailItemsRegistratorInjectable = getInjectable({
+  id: "extension-catalog-entity-detail-items-registrator",
+
+  instantiate:
+    (di) => (extension: LensRendererExtension, installationCounter) => {
+      pipeline(
+        extension.catalogEntityDetailItems.map((registration, index) =>
+          getInjectable({
+            id: `catalog-entity-detail-item-${index}-from-${extension.sanitizedExtensionId}-instance-${installationCounter}`,
+
+            instantiate: () => ({
+              apiVersions: registration.apiVersions,
+              kind: registration.kind,
+
+              components: {
+                Details: registration.components.Details,
+              },
+
+              orderNumber: -registration.priority || -50,
+              extension,
+            }),
+
+            injectionToken: catalogEntityDetailItemInjectionToken,
+          }),
+        ),
+
+        forEach(di.register),
```

Comment:
Why is this better than using `computed` directly? Now we have a whole bunch of instantiates every time any single extension gets added or removed or adds an item.

Reply:
`computed` has limitations. We noticed those during refactoring of routes. `computed` is especially problematic when there is `React` components inside. It's very easy to create new instance of a component every time.

`registrator`-pattern is something that we are trying for couple reasons: 
1. Extensions can just register more implementations of an injection token (use places don't need to prepare for it)
2. We have full control of given registrator in single place. Which means that we can keep use places flexible and create adapters for Extension API whenever needed without introducing breaking changes.

There is still some downsides in `registator`-pattern, most critical ones are the usage of installation counter and that we need to filter results of `injectMany` based on enabled extensions. This is however something that we are actively solving and the end goal is that you only need to `injectMany` and get results as reactive.

Reply:
Note: This is still something that we are trying out to see if it carries us over the finish line or not.

Reply:
> computed is especially problematic when there is React components inside. It's very easy to create new instance of a component every time.

What do you mean by this? Wouldn't some form of shallow equality or reference marker be the solution then?

Reply:
For example: 

In case of pages we have logic that makes page component observable if page params are defined. It also provides page params as props for the component. This requires that we introduce higher-order component for Page component if page params are present. This higher-order component can't be created inside `computed` since that would cause React to treat it as a new component (re-mount) every time that the value of computed recomputes.

```typescript
computed(() => {
  return pipeline({
    extensions,
    flatMap(extension => extension.clusterPages),

    map(pageRegistration => ({
      ...pageRegistration,
      
      components: {
        ...pageRegistration.components,
        
        ...(pageRegistration.params ? {
          Page: observer(() => (<pageRegistration.components.Page />))
        } : {})
      }
    });
  });
});
```

Notice that I don't remember the exact PageRegistration API so property names could be wrong.

Reply:
Why do we need to do that? I would argue that since the page dereferences computed values, if it is not marked as observable itself, that is a bug or at least wrong and should be fixed. Locality of use is key here.

Reply:
I completely agree with you on that. We shouldn't make the component `observer` in behind the scenes. However, that's how current implementation works and is exposed in Extension API so we can't make the change without introducing breaking change.  

Anyway, that's just one requirement. There might be other requirements for higher-order components which cause the same exact issue. E.g. we add `TabLayout` higher order component to page components if they have siblings.

Reply:
We are currently testing the hypothesis that Extensions register new implementations of an `injectionToken` which then allows us to treat extensions as first-class citizens and therefore introduce a lot simplicity. Imagine that everything is built in a way that if we that we want to expose something to Extension API, we just expose the interface for injection token. In that world, we have:

1. One place that registers all implementations of `injectionToken` from extension (extension is enabled)
2. One place that deregisters all injectables that was registered because of extension was enabled (extension is disabled)

In that way we can use `di.injectMany` (or something similar) safely everywhere we want to have all implementations and we don't need to worry about enabled extensions etc. Also `registrator` will be consolidated to one place. 

Treat `registrator` as an adapter for Extension API 1.0 -> Extension API 2.0, which allows us to experiment with requirements of an interface before we actually expose the interface itself to extensions.